File: archetype-common/src/main/java/com/alo7/archetype/common/mapper/primary/UserMapper.java
// Copyright 2019 Alo7 Inc. All rights reserved.

package com.alo7.archetype.common.mapper.primary;

import com.alo7.archetype.base.mybatis.crud.BasicCrudMapper;
import com.alo7.archetype.base.mybatis.crud.MapperTable;
import com.alo7.archetype.common.entity.primary.User;

import java.util.List;

/**
 * @author Kelin Tan
 */
@MapperTable(value = "user", columns = "id,user_name")
public interface UserMapper extends BasicCrudMapper<User> {
    List<User> findByName(String name);
}

File: archetype-demo-database/src/main/java/com/kelin/archetype/database/mapper/primary/UserMapper.kt
// Copyright 2020 Kelin Inc. All rights reserved.

package com.kelin.archetype.database.mapper.primary

import com.kelin.archetype.database.MapperTable
import com.kelin.archetype.database.entity.primary.User
import com.kelin.archetype.database.mybatis.crud.BaseCrudMapper
import org.apache.ibatis.annotations.Select

/**
 * @author Kelin Tan
 */
@MapperTable(value = UserMapper.TABLE, columns = UserMapper.COLUMNS)
interface UserMapper : BaseCrudMapper<User> {
    @Select("SELECT $COLUMNS FROM $TABLE WHERE user_name = #{name}")
    fun findByName(name: String): List<User>

    companion object {
        const val TABLE = "user"
        const val COLUMNS = "id,user_name"
    }
}